### ISSUE TITLE
twofactor code input request

### ISSUE THREAD

**Issue by iGoreSchmidt** [rg3/youtube-dl#5868]
If i provide only -u parameter with username, app asks to input password, but if then youtube required two factor authentication- there no request for user to input key generated by Google Authenticator app.

It will be great to have a possibility to input TFA-code on-demand.

THANKS

**Comment by vxbinaca** [comment]
Couldn't you make an app-specific password for youtube-dl?

**Comment by Saklad5** [comment]
This issue should be closed, as there is now a two-factor code option.

**Comment by nyuszika7h** [comment]
@vxbinaca No, youtube-dl doesn't support app-specific passwords, because it's not supported in browsers either, and obviously there is no official YouTube API to download videos without any sort of DRM, so youtube-dl pretends to be a browser.

Though 2FA support for YouTube is broken right now unless you use `--cookies` (see #13050), this issue can be closed as it has been implemented since it was opened and there is no need to have multiple open issues on the same topic.

**Comment by vxbinaca** [comment]
I concur. Cookies is a cleaner solution. Asking for a close as well.
